### ISSUE TITLE
HTML Parsing : inside script element - block

### ISSUE THREAD

**Issue by raphpell** [ScintillaOrg/lexilla#47]
Blocks not working correctly in SCRIPT html element.

Steps to Reproduce the Issue

 1- type <SCRIPT ></SCRIPT >
 2-   type <SCRIPT ></SCRIPT >
 3-  type JS Code in the second element

Expected Behavior
Blocks works properly (expanding, reducing js code)

Actual Behavior
Blocks don't works properly

**Comment by nyamatongwe** [comment]
This worked for me. Please provide an example file. Which application are you using?

**Comment by ArkadiuszMichalski** [comment]
```html
<html>
<head></head>
<body>

<script src="src/myjsfile.je"></script><!-- Delete this line tu reduce myFunction block -->
<script>
var myFunction = function(){
    /* I can't reduce this block - block not detected */
}
</script
>
</body>
</html>

```

Adding one or two new lines before `myFunction` (depending on the cursor position) fixes it too. When I remove these new lines it also works up to some point (here `<script>` opening tag).

**Comment by nyamatongwe** [comment]
This appears to be a minimal case.
~~~html
<script src=""></script>
<script>
{
};
</script>
~~~

Turning on visible line state (SciTE: fold.flags=128) shows that 3rd digit (from right) is 0 instead of 1 ('11', instead of '111') showing that clientScript is eScriptNone(0) instead of eScriptJS(1). The folding logic ignores '{' for eScriptNone. Remove the `src` attribute and its good. So, tracking the current client script needs some work.

**Comment by zufuliu** [comment]
it seems caused by `segIsScriptingIndicator()`:
https://github.com/ScintillaOrg/lexilla/blob/master/lexers/LexHTML.cxx#L98

`<script src=""></script>` yields `eScriptNone` due to the `src` attribute.
```cpp
	if (strstr(s, "src"))	// External script
		return eScriptNone;
```
the second script without attributes also yields `eScriptNone`:
```cpp
	return prevValue;
```

**Comment by zufuliu** [comment]
> the second script without attributes also yields `eScriptNone`:

Which is wrong, second `script` tag has no attributes so `segIsScriptingIndicator()` is not called.

This problem can be fixed with following changes:
```diff
diff --git a/lexers/LexHTML.cxx b/lexers/LexHTML.cxx
index 44f75db8..b615e944 100644
--- a/lexers/LexHTML.cxx
+++ b/lexers/LexHTML.cxx
@@ -1927,8 +1927,10 @@ void SCI_METHOD LexerHTML::Lex(Sci_PositionU startPos, Sci_Position length, int
 			if (!setAttributeContinue.Contains(ch)) {
 				if (inScriptType == eNonHtmlScript) {
 					const int scriptLanguagePrev = scriptLanguage;
-					clientScript = segIsScriptingIndicator(styler, styler.GetStartSegment(), i - 1, scriptLanguage);
-					scriptLanguage = clientScript;
+					scriptLanguage = segIsScriptingIndicator(styler, styler.GetStartSegment(), i - 1, scriptLanguage);
+					if (scriptLanguage != eScriptNone) { // not external script
+						clientScript = scriptLanguage;
+					}
 					if ((scriptLanguagePrev != scriptLanguage) && (scriptLanguage == eScriptNone))
 						inScriptType = eHtml;
 				}
```

**Comment by nyamatongwe** [comment]
That change fixes the problem and I haven't yet seen any regressions.

**Comment by zufuliu** [comment]
Attached a more correct and complex patch:
[html-clientscript-1222.zip](https://github.com/ScintillaOrg/lexilla/files/7762752/html-clientscript-1222.zip)

Sine script language can't be detected from attribute names, so removed `segIsScriptingIndicator()` from `SCE_H_ATTRIBUTE` case, instead changed `classifyAttribHTML()` to returns the attribute type and moved script language detection into `SCE_H_DOUBLESTRING` and `SCE_H_SINGLESTRING` cases for `type` or `language` script attribute as described at https://html.spec.whatwg.org/multipage/scripting.html#script-processing-model.

> 
> 6. If the element has no src attribute, and source text is the empty string, then return. The script is not executed.
> 
> 7. If the element is not connected, then return. The script is not executed.
> 
> 8. If either:
> 
>     the script element has a type attribute and its value is the empty string, or
>     the script element has no type attribute but it has a language attribute and that attribute's value is the empty string, or
>     the script element has neither a type attribute nor a language attribute, then
> 
> ...let the script block's type string for this script element be "text/javascript".
> 
> Otherwise, if the script element has a type attribute, let the script block's type string for this script element be the value of that attribute.
> 
> Otherwise, the element has a non-empty language attribute; let the script block's type string for this script element be the concatenation of the string "text/" followed by the value of the language attribute.

So the code will fails if  `script` tag has non-empty text but with a `src` attribute.

**Comment by zufuliu** [comment]
Updated patch to always treat inner text of `script` tag as script.
[html-script-1229.zip](https://github.com/ScintillaOrg/lexilla/files/7789193/html-script-1229.zip)

test case
```js
<script src="">
// comment
</script>
<script>
{
};
</script>
```

**Comment by nyamatongwe** [comment]
Issue now fixed.

**Comment by nyamatongwe** [comment]
Included in 5.1.5 release.
